kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/FilterHandler.java

```diff
@@ -127,11 +131,30 @@ else if (msg instanceof DecodedResponseFrame<?> decodedFrame) {
                 });
             }
             else {
-                throw new IllegalStateException("Unexpected message reading from upstream:  " + msg);
+                throw new IllegalStateException("Filter '" + filterAndInvoker.filterName() + "': Unexpected message reading from upstream: " + msgDescriptor(msg));
             }
         }
     }
 
+    /**
+     * @param obj A message
+     * @return A descriptor for the message (for logging purposes). Does not include the message contents.
+     */
+    static String msgDescriptor(@Nullable Object obj) {
```

Comment:
I realised that we were logging message contents (if only at debug). I don't think we should do that.

Reply:
Almost suggest we should have a custom exception which takes the message and filterName as args so this sort of formatting decision can be made when logging. 

That said your probably right we shouldn't be logging whole messages.